### PR TITLE
feat: add NB showing Lagrange pt computation (CR3BP)

### PR DESCRIPTION
**Pull request by DhruvJ22**
The notebook employs the celestial objects parameters defined and exposed by OTSK to compute the 5 Lagrange points of the Earth-Moon system modeled through CR3BP. The locations of the 5 points are computed with respect to the barycenter of the Earth-Moon system, expressed in a rotating reference frame.

### CODE REVIEW

**Comment by DhruvJ22** [comment]
@vishwa2710 and @kyle-cochran, pinging in case the repo is not configured to notify the owners. Apologies, I forgot to add you as the reviewer.

**Comment by vishwa2710** [comment]
Hi @DhruvJ22 , this is looks great!
- Can you update the MR title to follow [conventional commit patterns ](https://www.conventionalcommits.org/en/v1.0.0/). In this case, it would be `feat: add NB showing Lagrange pt computation (CR3BP)`

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    "# Calculate the average distance between p1 and p2 (over a lunar cycle ~ 28 days)\n",
+    "p1 = Earth.default()\n",
+    "p2 = Moon.default()\n",
+    "\n",
+    "# Define a common frame\n",
+    "frame = Frame.ITRF()\n",
+    "\n",
+    "def get_EM_dist_at_epoch(p1, p2, day):\n",
+    "\n",
+    "    instant = Instant.date_time(DateTime(2000, 1, 1+day, 0, 0, 0), Scale.UTC)\n",
```
Comment (vishwa2710) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb:114]:
```suggestion
    "    instant = Instant.J2000() + Duration.days(float(day))\n",
```

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    "\n",
+    "    # Calculate the distance between the two positions\n",
+    "    distance = np.linalg.norm(p1_position - p2_position)\n",
+    "    return distance\n",
+    "\n",
+    "# Distances over a lunar cycle\n",
+    "distances = [get_EM_dist_at_epoch(p1, p2, day) for day in range(28)]\n",
+    "\n",
+    "# Average distance in km\n",
+    "lstar = np.average(np.array(distances)/1000)\n",
```
Comment (vishwa2710) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb:128]:
```suggestion
    "lstar = np.average(np.array(distances)/1000.0)\n",
```

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+     "output_type": "stream",
+     "text": [
+      "P1 location wrt barycenter: [-4682.908336745254 0.0 0.0] km\n",
+      "P1 location wrt barycenter: [380723.12362949521 0.0 0.0] km\n"
+     ]
+    }
+   ],
+   "source": [
+    "pos_p1 = np.array([-1*mu,0,0])*lstar # unit: km, value < radius of the Earth\n",
+    "pos_p2 = np.array([1-1*mu,0,0])*lstar # unit: km\n",
```
Comment (vishwa2710) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb:159]:
```suggestion
    "pos_p1 = np.array([-1 * mu, 0, 0]) * lstar # unit: km, value < radius of the Earth\n",
    "pos_p2 = np.array([1 - 1 * mu, 0, 0]) * lstar # unit: km\n",
```

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pos_l4 = np.array([0.5-mu,3**0.5/2,0])\n",
+    "pos_l5 = np.array([0.5-mu,-3**0.5/2,0])"
```
Comment (vishwa2710) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb:181]:
```suggestion
    "pos_l4 = np.array([0.5 - mu, 3**0.5 / 2, 0])\n",
    "pos_l5 = np.array([0.5 - mu, -3**0.5 / 2, 0])"
```

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+    "        [1, 3 - mu, 3 - 2 * mu, - 1 * mu, -2 * mu, - 1 * mu],\n",
+    "        [1, 2 + mu, 1 + 2 * mu, mu - 1, 2 * mu - 2, -1 + mu],\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "def get_real_root(poly):\n",
+    "    roots = np.roots(poly)\n",
+    "    for root in roots:\n",
+    "        if np.imag(root)==0:\n",
+    "            return np.real(root)\n",
```
Comment (vishwa2710) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb:225]:
```suggestion
    "def get_real_root(poly):\n",
    "    roots = np.roots(poly)\n",
    "    for root in roots:\n",
    "        if np.isreal(root):\n",
    "            return np.real(root)\n",
```

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+    "\n",
+    "lag_loc[0, 0] = 1 -1*mu - get_real_root(f_lib[0,:])\n",
+    "lag_loc[1, 0] = 1 -1*mu + get_real_root(f_lib[1,:])\n",
+    "lag_loc[2, 0] = -1*mu - get_real_root(f_lib[2,:])\n",
```
Comment (vishwa2710) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb:229]:
```suggestion
    "lag_loc[0, 0] = 1 -1 * mu - get_real_root(f_lib[0,:])\n",
    "lag_loc[1, 0] = 1 -1 * mu + get_real_root(f_lib[1,:])\n",
    "lag_loc[2, 0] = -1 * mu - get_real_root(f_lib[2,:])\n",
```

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+    "\n",
+    "# Store location of L4 and L5 in lag_loc\n",
+    "lag_loc[3,:] = pos_l4\n",
+    "lag_loc[4,:] = pos_l5\n",
+    "\n",
+    "# Change units from non-dimensional to km, w.r.t barycenter\n",
+    "lag_loc = lag_loc*lstar\n",
```
Comment (vishwa2710) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb:236]:
```suggestion
    "lag_loc = lag_loc * lstar\n",
```

**Review by vishwa2710 — COMMENTED** [review]
Looks great man! Just some nitpicks.

File: notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+      "P1 location wrt barycenter: [-4682.908336745254 0.0 0.0] km\n",
+      "P1 location wrt barycenter: [380723.12362949521 0.0 0.0] km\n"
+     ]
+    }
+   ],
+   "source": [
+    "pos_p1 = np.array([-1*mu,0,0])*lstar # unit: km, value < radius of the Earth\n",
+    "pos_p2 = np.array([1-1*mu,0,0])*lstar # unit: km\n",
+    "print(f'P1 location wrt barycenter: {pos_p1} km')\n",
+    "print(f'P1 location wrt barycenter: {pos_p2} km')"
```
Comment (kyle-cochran) [inline notebooks/Astrodynamics/Orbit Computation/Lagrange Point Computation CR3BP.ipynb]:
```suggestion
    "print(f'P2 location wrt barycenter: {pos_p2} km')"
```

typo?

**Comment by DhruvJ22** [comment]
Thank you for suggesting the changes! I have incorporated all of them.

**Review by kyle-cochran — APPROVED** [review]
Really awesome! One nitpick from me, but looks great!

**Review by vishwa2710 — APPROVED** [review]

**Comment by vishwa2710** [comment]
Looks great @DhruvJ22 ! Congratulations and thank you for your first contribution to OSTk! Look forward to more. Stay in touch on the discord.

**Comment by DhruvJ22** [comment]
Thank you, @kyle-cochran and @vishwa2710! I think the PR is ready to be merged.

**Pull request merged**